engineering/frontend.md

```diff
@@ -27,6 +35,11 @@ We currenty don't have a strict rule for writing Javascript as it would limit de
 - Prefer functional & immutable array methods .ie `map`/`filter`/`reduce`/`some`/`every` over any types of mutable `for` loop
 - Prefer "return early" coding style, more about it [here](https://medium.com/@matryer/line-of-sight-in-code-186dd7cdea88)
 - Don't obsess over performance of code, obsess over making it clear
+- Use pure functions where possible to make them testable and modular.
+- Avoid mutating varibles and the `let` keyword.
+- React Components should be stateless functional components where possible.
+- Avoid classes and stateful modules where possible.
+- Avoid React lifecyle hooks.
```

Comment:
why do we want to "avoid" platform features here?

Reply:
@anduong could you elaborate?